R/tmb.R

```diff
@@ -94,11 +95,25 @@ h_mmrm_tmb_formula_parts <- function(formula) {
     stop(
       "Covariance structure must be of the form `",
       cov_term[[1]],
-      "(time|subject)`"
+      "(time|(group/)subject)`"
     )
   }
   visit_term <- cov_content[[2L]]
-  subject_term <- cov_content[[3L]]
+  if (identical(length(cov_content[[3L]]), 1L)) {
+    subject_term <- cov_content[[3L]]
+    group_var <- NULL
+  } else if (identical(length(cov_content[[3L]]), 3L) && identical(as.character(cov_content[[3L]][[1L]]), "/")) {
+    subject_term <- cov_content[[3L]][[3L]]
+    group_term <- cov_content[[3L]][[2L]]
+    assert_true(identical(length(group_term), 1L))
+    group_var <- deparse(group_term)
+  } else {
+    stop(
+      "Covariance structure with grouping must be of the form `",
+      cov_term[[1]],
+      "(time|group/subject)`"
+    )
+  }
```

Comment:
since this is quite complex, would be good to move to small helper function e.g. `h_mmrm_tmb_extract_group()` - that helps with readability and with unit testing